### ISSUE TITLE
Is there a recommended way to pass additional information to the decoder?

### ISSUE THREAD

**Issue by grudloff** [jdb78/pytorch-forecasting#1713]
I am working with a use case where I have relevant information about each group (they are cliend_ids and I have info about location, age, etc). Is there a recommended way to pass this information to the decoder?

I think the most straightforward way would be to create a new class that inherits from the model I am using (RecurrentNetwork in my case), fetch this information based on the group, and pass it together with the rest to the network.

Any input would be greatly appreciated!

**Comment by grudloff** [comment]
Ultimately, I just extended the dataset with the additional client information as the dataset is small and this was feasible. Then I passed these new features as known categorical/numerical values in the dataloader.

Having said that, I believe it would be nice to add a lazy loading option for this use case, as a normalized dataset is a common pattern.
